apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/peer/PeerLoaderThread.java

```diff
@@ -16,40 +17,48 @@
 /**
  * @author alukin@gmail.com
  */
+@Slf4j
 class PeerLoaderThread implements Runnable {
     private static final Logger LOG = LoggerFactory.getLogger(PeerLoaderThread.class);
     private final List<String> defaultPeers;
     private final List<Future<String>> unresolvedPeers;
     private final Set<PeerDb.Entry> entries = new HashSet<>();
     private final TimeService timeService;
     private final PeersService peersService;
-    private PeerDb peerDb;
+    private final PeerDb peerDb;
 
-    public PeerLoaderThread(List<String> defaultPeers, List<Future<String>> unresolvedPeers, TimeService timeService, PeersService peersService) {
+    public PeerLoaderThread(List<String> defaultPeers,
+                            List<Future<String>> unresolvedPeers,
+                            TimeService timeService,
+                            PeersService peersService) {
         this.defaultPeers = defaultPeers;
         this.unresolvedPeers = unresolvedPeers;
         this.timeService = timeService;
         this.peersService = peersService;
+        this.peerDb = peersService.getPeerDb();
     }
 
     @Override
     public void run() {
         LOG.trace("'Peer loader': thread starting...");
-        if (peerDb == null) {
-            peerDb = CDI.current().select(PeerDb.class).get();
+        if (this.peerDb == null) {
+//            peerDb = CDI.current().select(PeerDb.class).get();
+            String error = "ERROR, the peerDb instance was not initialized inside peerService";
+            log.error(error);
+            throw new RuntimeException(error);
         }
         final int now = timeService.getEpochTime();
         peersService.wellKnownPeers.forEach((address) -> {
             PeerAddress pa = new PeerAddress(address);
             entries.add(new PeerDb.Entry(pa.getAddrWithPort(), 0, now));
         });
-        if (peersService.usePeersDb) {
+        if (PeersService.usePeersDb) {
```

Comment:
I guess, that all those options from PeersService (userPeersDb, savePeers, etc) should not be static

Reply:
It seems to me that doesn't matter here.
Most of variables are initialized inside PeersService -> public void init() {......) and later are used as 'read-only' fields.